infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/utils/LogReplicationConfigManager.java

```diff
@@ -41,98 +64,251 @@
+                    corfuStore.openTable(CORFU_SYSTEM_NAMESPACE,
+                            LR_MODEL_METADATA_TABLE_NAME,
+                            ClientDestinationInfoKey.class,
+                            DestinationInfoVal.class,
+                            null,
+                            TableOptions.fromProtoSchema(ClientRegistrationInfo.class));
+                } catch (InvocationTargetException | NoSuchMethodException | IllegalAccessException e) {
+                    log.error("Cannot open client config tables, retrying.", e);
+                    throw new RetryNeededException();
+                }
+                return null;
+            }).run();
+        } catch (InterruptedException e) {
+            log.error("Failed to open client config tables", e);
+            throw new UnrecoverableCorfuInterruptedError(e);
+        }
+    }
+
+    /**
+     * Generate LogReplicationConfig for all given sessions based on there replication model.
+     *
+     * @param sessions set of sessions for which to generate config
+     */
+    public void generateConfig(Set<LogReplicationSession> sessions) {
+        sessions.forEach(session -> {
+                switch (session.getSubscriber().getModel()) {
+                    case FULL_TABLE:
+                        log.debug("Generating FULL_TABLE config for session {}",
+                                TextFormat.shortDebugString(session));
+                        generateFullTableConfig(session);
+                        break;
+                    case LOGICAL_GROUPS:
+                        log.debug("Generating LOGICAL_GROUP config for session {}",
+                                TextFormat.shortDebugString(session));
+                        generateLogicalGroupConfig(session);
+                        break;
+                    default: break;
+                }
+        });
+    }
+
+    private void generateLogicalGroupConfig(LogReplicationSession session) {
         Map<UUID, List<UUID>> streamToTagsMap = new HashMap<>();
+        Set<String> streamsToReplicate = new HashSet<>();
+        // Check if the local cluster is the Sink for this session. Sink side will honor whatever Source side send
+        // instead of relying on groupSinksMap to filter the streams to replicate.
+        boolean isSink = session.getSinkClusterId().equals(localClusterId);
+        Map<String, Set<String>> logicalGroupToStreams = new HashMap<>();
+        if (!isSink) {
+            groupSinksMap.entrySet().stream()
+                    .filter(entry -> entry.getValue().contains(session.getSinkClusterId()))
+                    .map(Map.Entry::getKey)
+                    .forEachOrdered(group -> {
+                        logicalGroupToStreams.put(group, new HashSet<>());
+                    });
+            log.debug("Targeted logical groups={}", logicalGroupToStreams.keySet());
+        }
 
         registryTableEntries.forEach(entry -> {
```

Comment:
This in-memory set may be stale when the config must be generated.  It must be rebuilt here.